web/public/electron.js

```diff
@@ -0,0 +1,35 @@
+const electron = require('electron');
+const app = electron.app;
```

Comment:
Does it matter if the PR is big?  I thought you wanted to get this done quickly.  I think electron is valuable and might ultimately be how it's distributed.  It can be removed in 10 seconds if needed.  But I spent time getting it working and to delete it and re add it later seems like a waste of time.

Reply:
I think we're adding-on a lot of stuff to this PR -- prefer having it in smaller chunks where possible. 
It's actually taking a lot longer to review because there's so much stuff going on.
here are [some of the reasons why I prefer smaller PRs](https://gist.github.com/indiesquidge/cc598dbd889d5e0511b0183e68912342).

I would have ideally liked the server package to be part of the first PR and the web package to be part of the second one but since it's all in one now it's ok but we shouldn't be adding more to it, which only makes _everything_ harder to review.